crates/extrinsics/src/lib.rs

```diff
@@ -20,6 +20,7 @@ mod error;
 mod events;
 mod instantiate;
 mod remove;
+pub mod runtime_api;
```

Comment:
I'm not sure about exposing this entire module, there's a lot in there and it can easily cause breaking changes any time we update the metadata. We could

1. Move the code which fetches the `ContractInfo` into the `extrinsics` crate and export that.
2. Only export `ContractInfo` and the function used to fetch it.

What do you think?

Reply:
`fetch_contract_info` takes `InfoCommand` as a parameter. 
The `extrinsics` crate doesn't have access to it.
As far as I am concerned, we can't use it without having cyclic dependencies.

Is there another way around this?

Reply:
It only uses `&self.contract` from `InfoCommand` so could just pass that as a parameter instead.

For the return type could even hide the `ContractInfo` type and return `InfoToJson` so we control the API instead of the codegen.

Reply:
> It only uses &self.contract from InfoCommand so could just pass that as a parameter instead.

Yes that's fine

but for the return type. `InfoToJson` is defined in `cargo_contract` crate so it can't be the return type of the new function in the `extrinsics` crate.
and even if we have it, how can we cast `ContractInfo` to `InfoToJson`?

Reply:
Move `InfoToJson` to `extrinsics`, maybe just rename it to `ContractInfo`.

> and even if we have it, how can we cast ContractInfo to InfoToJson?

Just build the instance as we already do:

```rust
                    let convert_trie_id = hex::encode(info_result.trie_id.0);
                    let info_to_json = InfoToJson {
                        trie_id: convert_trie_id,
                        code_hash: info_result.code_hash,
                        storage_items: info_result.storage_items,
                        storage_item_deposit: info_result.storage_item_deposit,
                    };
```

Reply:
Thanks for the detailed answer!
Let me know if this is now resolved.